prometheus-to-sd/config/source_config_test.go

```diff
@@ -160,13 +188,38 @@ func TestParseSourceConfig(t *testing.T) {
 				},
 			},
 			SourceConfig{
-				Component:     "testComponent",
-				Protocol:      "http",
-				Host:          "localhost",
-				Port:          8080,
-				Path:          defaultMetricsPath,
-				MetricsPrefix: "container.googleapis.com/newPrefix",
-				PodConfig:     NewPodConfig(podId, namespaceId, "", "", ""),
+				Component: "testComponent",
+				Protocol:  "http",
+				Host:      "localhost",
+				Port:      8080,
+				Path:      defaultMetricsPath,
+				//AuthConfig:           userAuthConfig,
+				MetricsPrefix:        "container.googleapis.com/newPrefix",
+				PodConfig:            NewPodConfig(podId, namespaceId, "", "", ""),
+				WhitelistedLabelsMap: emptyWhitelistedLabelsMap,
+			},
+		},
+		{
+			flags.Uri{
+				Key: "testComponent",
+				Val: url.URL{
+					Scheme:   "http",
+					Host:     "hostname:1234",
+					Path:     defaultMetricsPath,
+					RawQuery: "whitelistedLabels=container_name:test_container|component_id:test_id",
+				},
+			},
+			SourceConfig{
+				Component: "testComponent",
+				Protocol:  "http",
+				Host:      "hostname",
+				Port:      1234,
+				Path:      defaultMetricsPath,
+				//AuthConfig: userAuthConfig,
+				WhitelistedLabelsMap: map[string]map[string]bool{
+					"container_name": {"test_container": true},
+					"component_id":   {"test_id": true}},
+				PodConfig: NewPodConfig(podId, namespaceId, "", "", ""),
```

Comment:
Could you add one more test case, with multiple whitelisted values for one label?

Reply:
Thanks, added another test case.